org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/hover/LSPTextHover.java

```diff
@@ -61,6 +62,8 @@ public class LSPTextHover implements ITextHover, ITextHoverExtension {
 	private static final MarkupParser MARKDOWN_PARSER = new MarkupParser(new MarkdownLanguage(true));
 	private static final int GET_TIMEOUT_MS = 1000;
 
+	private LanguageServerDocumentExecutor lastExecutor;
```

Comment:
Also, this shows that with this proposal, the client needs to keep a reference to the executor instead of a reference to the request future. It's not a clear win.

Reply:
Indeed. It's not perfect.